pkg/selector/outputs/bubbletea.go

```diff
@@ -0,0 +1,282 @@
+			m.TableModel = m.TableModel.WithPageSize(newRowsPerPage)
+		}
+	}
+
+	// update table
+	var cmd tea.Cmd
+	m.TableModel, cmd = m.TableModel.Update(msg)
+
+	// update footer
+	controlsStr := lipgloss.NewStyle().Faint(true).Render(controlsString)
+	footerStr := fmt.Sprintf("Page: %d/%d | %s", m.TableModel.CurrentPage(), m.TableModel.MaxPages(), controlsStr)
+	m.TableModel = m.TableModel.WithStaticFooter(footerStr)
+
+	return m, cmd
+}
+
+// View is used by bubble tea to render the bubble tea model
+func (m BubbleTeaModel) View() string {
+	outputStr := strings.Builder{}
+
+	outputStr.WriteString(m.TableModel.View())
+	outputStr.WriteString("\n")
+
+	return outputStr.String()
+}
+
+// table creation helpers:
+
+// createRows creates a row for each instance type in the passed in list
+func createRows(instanceTypes []*instancetypes.Details) *[]table.Row {
+	rows := []table.Row{}
+
+	for _, instanceType := range instanceTypes {
+		none := "none"
+		hyperisor := instanceType.Hypervisor
+		if hyperisor == nil {
+			hyperisor = &none
+		}
+
+		cpuArchitectures := []string{}
+		for _, cpuArch := range instanceType.ProcessorInfo.SupportedArchitectures {
+			cpuArchitectures = append(cpuArchitectures, *cpuArch)
+		}
+
+		gpus := int64(0)
+		gpuMemory := int64(0)
+		gpuType := []string{}
+		if instanceType.GpuInfo != nil {
+			gpuMemory = *instanceType.GpuInfo.TotalGpuMemoryInMiB
+			for _, gpuInfo := range instanceType.GpuInfo.Gpus {
+				gpus = gpus + *gpuInfo.Count
+				gpuType = append(gpuType, *gpuInfo.Manufacturer+" "+*gpuInfo.Name)
+			}
+		}
+
+		onDemandPricePerHourStr := "-Not Fetched-"
+		spotPricePerHourStr := "-Not Fetched-"
+		if instanceType.OndemandPricePerHour != nil {
+			onDemandPricePerHourStr = "$" + formatFloat(*instanceType.OndemandPricePerHour)
+		}
+		if instanceType.SpotPrice != nil {
+			spotPricePerHourStr = "$" + formatFloat(*instanceType.SpotPrice)
+		}
+
+		newRow := table.NewRow(table.RowData{
+			colInstanceType:       *instanceType.InstanceType,
+			colVCPU:               *instanceType.VCpuInfo.DefaultVCpus,
+			colMemory:             formatFloat(float64(*instanceType.MemoryInfo.SizeInMiB) / 1024.0),
+			colHypervisor:         *hyperisor,
+			colCurrentGen:         *instanceType.CurrentGeneration,
+			colHibernationSupport: *instanceType.HibernationSupported,
+			colCPUArch:            strings.Join(cpuArchitectures, ", "),
+			colNetworkPerformance: *instanceType.NetworkInfo.NetworkPerformance,
+			colENI:                *instanceType.NetworkInfo.MaximumNetworkInterfaces,
+			colGPU:                gpus,
+			colGPUMemory:          formatFloat(float64(gpuMemory) / 1024.0),
+			colGPUInfo:            strings.Join(gpuType, ", "),
+			colODPrice:            onDemandPricePerHourStr,
+			colSpotPrice:          spotPricePerHourStr,
+		})
+
+		rows = append(rows, newRow)
+	}
+
+	return &rows
+}
+
+// createColumns creates columns based on the column key constants
+func createColumns() *[]table.Column {
```

Comment:
if you can de-dupe the logic in createRows by passing the data to the createRows func, you could pass the same data to createColumns and loop through the struct using tags (https://www.digitalocean.com/community/tutorials/how-to-use-struct-tags-in-go)

Reply:
I have de-duped the logic found in `TableOutputWide`, `createRows`, and `createColumns`. Your recommendation to use struct tags was extremely helpful, and said struct tags can now be found in `outputs.go` in the `wideColumnsData` struct.

Reply:
nice!